theano/gof/compilelock.py

```diff
@@ -101,7 +101,14 @@ def get_lock(lock_dir=None, **kw):
             if now - get_lock.start_time > config.compile.timeout/2:
                 lockpath = os.path.join(get_lock.lock_dir, 'lock')
                 _logger.info('Refreshing lock %s', str(lockpath))
-                refresh_lock(lockpath)
+                try:
+                    refresh_lock(lockpath)
+                except Exception:
+                    while get_lock.lock_is_enabled:
+                        release_lock()
```

Comment:
The lock can be recursive so this gets rid of all the levels of the lock.

But I think it's a bad idea to do this and also that this exception catching should go in refresh_lock().